client/src/context/GameContext.tsx

```diff
@@ -93,6 +96,7 @@ export const GameProvider: React.FC = ({ children }) => {
   const [stepIndex, setStepIndex] = useState(0);
   const [numElems, setNumElems] = useState(10);
   const [[min, max], setMinMax] = useState([0, 20]);
+  const [maxLevelSeen, setMaxLevelSeen] = useState<number>(0);
```

Comment:
nit: typescript is pretty good at interpreting types, don't need the `<number>` here. also just nicer for the sake of consistency with the states defined in the lines above.